src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -6,17 +6,81 @@ description: Motion and how it works in the NUbots codebase.
 slug: /system/subsystems/motion
 ---
 
-The motion subsystem contains modules that are responsible for walking, kicking, moving the head, and running scripts. They perform these motions by sending servo commands (called targets) to the motors.
+This page details motion modules within the NUbots main codebase, such as the walk engines and the Script Engine. The Script Engine is detailed on this page, while Script Runner and Script Tuner can be found on the [Behaviour](/system/subsystems/behaviour#script-runner) page.
+
+
+## Walk Engine
+
+Currently NUbots has two walk engines. One is an **open loop hybrid walk controller** and the other is an **open loop walk** based on **quintic splines**. These engines take care of the mathematics and control involved in taking steps without falling over. The walk engines do not plan where to walk to, only how to walk.
+
+A walk engine will trigger when a [**walkcommand**](https://github.com/NUbots/NUbots/blob/master/shared/message/motion/WalkCommand.proto) is [emitted](/system/foundations/nuclear#emit-statements). The walk command gives a vector [x, y, z] which represents:
+
+| Variable | Description                                                                                |
+| -------- | ------------------------------------------------------------------------------------------ |
+| x        | velocity in metres per second                                                              |
+| y        | velocity in metres per second                                                              |
+| z        | yaw in radians per second, positive is counter-clockwise. The 0 point is straight forward. |
```

Comment:
wondering if we should refer to yaw change per second as  `$\omega$` instead of `z` perhaps, like in physics stuff

Reply:
This is probs good for now since it's how it's described in the codebase and how Eigen perceives it. Once you've done that refactor you talked about the other day, can fix this up.

Reply:
There is a right=hand rule for this if we haven't already added it to the maths page
```suggestion
| z        | yaw in radians per second, positive is counter-clockwise around the positive z-axis. The 0 point is straight forward. |
```

![image](https://user-images.githubusercontent.com/2320984/124342474-981f4500-dc07-11eb-9337-3b48e928f467.png)